File: w06/d02/Sal/mta/mta.js
// CREATE LINE OBJECT 
function Line(line) {
  this.line = line;
  this.stops = [];
}

// LINE DISTANCE FUNCTION 
Line.prototype.distance = function(destination, origin) {
    return Math.abs(destination - origin);
}

// CREATE LINES
var n = new Line("n");
n.stops = ["Times Square", "34th", "N-28th", "23rd", "Union Square", "8th"];

var l = new Line("l");
l.stops = ["8th", "6th", "Union Square", "3rd", "1st"];

var s = new Line("s");
s.stops = ["Grand Central", "33rd", "S-28th", "23rd", "Union Square", "Astor Place"];

// CREATE SUBWAY SYSTEM OBJECT 
function SubwaySystem(name) {
  this.name = name;
  this.lines = [];
}

// CREATE NEW SUBWAY SYSTEM 
var mta = new SubwaySystem("mta");
mta.lines = [n, l, s];

// TAKE TRIP FUNCTION
SubwaySystem.prototype.takeTrip = function() {
  var originLine = prompt("What line do you want to get on at? N, L, or S?").toLowerCase();
  switch (originLine) {
    case "n":
      originLine = n;
      break; 
    case "l":
      originLine = l; 
      break;
    case "s":
      originLine = s;
      break;
    }

  var originStop = prompt("Stops to get on: " + originLine.stops);
  originStop = originLine.stops.indexOf(originStop);

  var destinationLine = prompt("Line to get off at: N, L, or S").toLowerCase();
  switch (destinationLine) {
  case "n":
    destinationLine = n;
    break; 
  case "l":
    destinationLine = l; 
    break;
  case "s":
    destinationLine = s;
    break;
  }

  var destinationStop = prompt("Stops to get off: " + destinationLine.stops);
  destinationStop = destinationLine.stops.indexOf(destinationStop);

  var unionOrigin = originLine.stops.indexOf("Union Square");
  var unionDestination = destinationLine.stops.indexOf("Union Square");

  if (originLine === destinationLine) {
    alert("You've traveled " + originLine.distance(destinationStop, originStop) + " stops!");
  } else if (originLine != destinationLine) {
    alert("You've traveled " + ( originLine.distance(unionOrigin, originStop) + 
    destinationLine.distance(unionDestination, destinationStop) )  + " stops!");
  }
}

File: w01/d03/Arun/mta.rb
mta = {n: ['ts', '34th', '28th-n', '23rd-n', 'us'], 
	   l: ['8th', '6th', 'us', '3rd', '1st'],
	   s: ['gc', '33rd', '28th-s', '23rd-s', 'us']}

puts "Which subway line would you like to enter (N or L)?"
	origin_line = gets.chomp.downcase
if origin_line == "n"
	puts mta[:n]
elsif origin_line == "l"
	puts mta[:l]
end
origin_line = origin_line.to_sym
puts "Which stop would you like to enter?"
origin_stop = mta[origin_line].index(gets.chomp.downcase)
puts "what line would you like to get off at?"
destination_line = gets.chomp.downcase
	if destination_line == "n"
		puts mta[:n]
	elsif destination_line == "l"
		puts mta[:l]
	end
destination_line = destination_line.to_sym
puts "Where would you like to get ofF?"
destination_stop = mta[destination_line].index(gets.chomp.downcase)

if origin_line = "n"

	puts "ride #{(destination_stop - origin_stop).to_i.abs} stops"
#elsif origin_line == "n" && destination_line == "l"
#	answer = ((4 - origin_stop).abs + (2 - destination_stop).abs)
#elsif origin_line == "l" && destination_line == "n"
#	answer = ((4 - origin_stop).abs + (2 - destination_stop).abs)

#puts "#{answer}"

